### ISSUE TITLE
Unknown variable error while querying Type

### ISSUE THREAD

**Issue by BennuFire** [neo4j-graphql/neo4j-graphql-js#489]
While using `2.14.2` , next type was created with a silly self reference.

```
type CONSUMER {
  original_id: ID!
  name: String
  description: String
  type: String
  close_flag: String
  stationTest: [CONSUMER] @cypher(statement: "return this")
}
```

Querying any info but stationTest works fine, but adding stationTest ends in with an internal server error:

```
query {
  CONSUMER(original_id : "43569202" ) {
    name
    description
    close_flag
    stationTest {
      name
    }
  }
}
```

>  "message": "Unknown variable `cONSUMER_stationTest`.",

It's interesting that adding a _first_ clause to the query, It works!!

PS: No, I don't really need that silly stationTest property.

**Comment by emech-en** [comment]
I have same issue with cypher directives.

**Comment by anakornk** [comment]
I have the same issue. Upgrading neo4j from v4.0.4 to v4.1.13  fixed the issue for me.
Should be related to this problem https://github.com/neo4j-graphql/neo4j-graphql-js/issues/445

**Comment by johnymontana** [comment]
Please upgrade to the latest Neo4j 4.x series release as I believe this has been resolved in the Cypher planner per #445
